src/main_thread/init/utils/initial_seek_and_play.ts

```diff
@@ -107,18 +107,37 @@ export default function performInitialSeekAndPlay(
       if (!isDirectfile || typeof startTime === "number") {
         const initiallySeekedTime =
           typeof startTime === "number" ? startTime : startTime();
-        if (initiallySeekedTime !== 0) {
+        if (initiallySeekedTime !== 0 && initiallySeekedTime !== undefined) {
           performInitialSeek(initiallySeekedTime);
         }
         waitForSeekable();
       } else {
         playbackObserver.listen(
           (obs, stopListening) => {
+            const initiallySeekedTime =
+              typeof startTime === "number" ? startTime : startTime();
+            if (
+              initiallySeekedTime === undefined &&
+              obs.readyState < 2 /* HAVE_CURRENT_DATA */
```

Comment:
You have https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/readyState#htmlmediaelement.have_current_data that allow to remove the need to have a comment here